common-rest/src/main/java/org/uniprot/api/rest/output/converter/AbstractUUWHttpMessageConverter.java

```diff
@@ -71,14 +70,33 @@ public boolean canRead(
     }
 
     @Override
-    protected MessageConverterContext<C> readInternal(
-            Class<? extends MessageConverterContext<C>> aClass, HttpInputMessage httpInputMessage) {
+    public MessageConverterContext<C> read(
+            Type type, Class<?> aClass, HttpInputMessage httpInputMessage) {
         return null;
     }
 
+    protected boolean validMediaType(MediaType mediaType) {
+        if (Utils.notNull(mediaType)) {
+            for (MediaType supportedMediaType : getSupportedMediaTypes()) {
+                if (mediaType.equals(supportedMediaType)) {
+                    return true;
+                }
+            }
+        }
+
+        // the following (taken from
+        // org.springframework.http.converter.AbstractHttpMessageConverter)
+        // can be simplified but this way is more readable.
+        if (mediaType == null || MediaType.ALL.equalsTypeAndSubtype(mediaType)) {
```

Comment:
Can you please help me to understand this bit, What is not clear for me is this mediaType == null.

I was thinking that this if would be 
if (mediaType != null && MediaType.ALL.equalsTypeAndSubtype(mediaType))

I was not able to spot a scenario that we write a content when we do not know the media type (mediaType == null)

Sorry, if this is a dummy question.